go/vt/vtgate/engine/aggregations.go

```diff
@@ -232,28 +258,51 @@ func (a *aggregatorSum) reset() {
 }
 
 type aggregatorScalar struct {
-	from    int
-	current sqltypes.Value
-	init    bool
+	from     int
+	current  sqltypes.Value
+	hasValue bool
 }
 
-func (a *aggregatorScalar) add(row []sqltypes.Value) error {
-	if !a.init {
+func (a *aggregatorScalar) add(row []sqltypes.Value, _ *evalengine.ExpressionEnv) error {
+	if !a.hasValue {
 		a.current = row[a.from]
-		a.init = true
+		a.hasValue = true
 	}
 	return nil
 }
 
-func (a *aggregatorScalar) finish() sqltypes.Value {
-	return a.current
+func (a *aggregatorScalar) finish(env *evalengine.ExpressionEnv) (sqltypes.Value, error) {
+	return a.current, nil
 }
 
 func (a *aggregatorScalar) reset() {
 	a.current = sqltypes.NULL
-	a.init = false
+	a.hasValue = false
+}
+
+type aggregatorConstant struct {
+	expr evalengine.Expr
+}
+
+func (*aggregatorConstant) add([]sqltypes.Value, *evalengine.ExpressionEnv) error {
+	return nil
+}
+
+func (a *aggregatorConstant) finish(env *evalengine.ExpressionEnv) (sqltypes.Value, error) {
+	return eval(env, a.expr)
+}
+
+func eval(env *evalengine.ExpressionEnv, eexpr evalengine.Expr) (sqltypes.Value, error) {
+	v, err := env.Evaluate(eexpr)
+	if err != nil {
+		return sqltypes.Value{}, err
+	}
+
+	return v.Value(collations.Unknown), nil
```

Comment:
Is this correct? I think this should be the default collation from the expression env?